[DOC-10001] Add upgrade content for 24.1 pre-production previews

mdlinville opened the pull request:
[DOC-10001] Add upgrade content for 24.1 pre-production previews

### Previews
[releases/cloud.md](https://deploy-preview-18427--cockroachdb-docs.netlify.app/docs/releases/cloud.html)
[src/current/cockroachcloud/upgrade-to-v24.1.md](https://deploy-preview-18427--cockroachdb-docs.netlify.app/docs/cockroachcloud/upgrade-to-v24.1.html)
[src/current/cockroachcloud/upgrade-to-v23.2.md](https://deploy-preview-18427--cockroachdb-docs.netlify.app/docs/cockroachcloud/upgrade-to-v23.2.html)

mdlinville commented on src/current/cockroachcloud/upgrade-to-v24.1.md:
```diff
@@ -0,0 +1,174 @@
+This [testing release]({% link releases/index.md %}#release-naming) is not qualified for production environments and not eligible for support or uptime SLA commitments.
+{{site.data.alerts.end}}
+
+An [Org Administrator]({% link cockroachcloud/authorization.md %}#org-administrator) can upgrade your CockroachDB {{ site.data.products.dedicated }} cluster from the CockroachDB {{ site.data.products.cloud }} Console. This page guides you through the process of upgrading.
```
```suggestion
An [Org Administrator]({% link cockroachcloud/authorization.md %}#org-administrator) can upgrade your CockroachDB {{ site.data.products.dedicated }} cluster from the CockroachDB {{ site.data.products.cloud }} Console. This page shows how to upgrade a CockroachDB {{ site.data.products.dedicated }} cluster to {{ page.pre_production_preview_version }} for testing and experimentation.
```

mdlinville commented on src/current/cockroachcloud/upgrade-to-v24.1.md:
```diff
@@ -0,0 +1,174 @@
+Upgrading from {{ page.prev_version }} to {{ page.pre_production_preview_version }} is a major-version upgrade. Upgrading a CockroachDB {{ site.data.products.dedicated }} cluster to a new major version is opt-in. Before proceeding, review the CockroachDB {{ site.data.products.cloud }} [CockroachDB Cloud Upgrade Policy](https://cockroachlabs.com/docs/cockroachcloud/upgrade-policy#pre-production-pr ...
+{{site.data.alerts.end}}
+
+This page shows how to upgrade a CockroachDB {{ site.data.products.dedicated }} cluster to {{ page.pre_production_preview_version }} for testing and experimentation.
```
```suggestion
```

mdlinville commented on src/current/cockroachcloud/upgrade-to-v24.1.md:
```diff
@@ -0,0 +1,174 @@
+
+This page shows how to upgrade a CockroachDB {{ site.data.products.dedicated }} cluster to {{ page.pre_production_preview_version }} for testing and experimentation.
+{% else %}
+Now that [CockroachDB {{ page.page_version }}](https://www.cockroachlabs.com/docs/releases/ {{ page.page_version }}) is available, an [Org Administrator]({% link cockroachcloud/authorization.md %}#org-administrator) can upgrade your CockroachDB {{ site.data.products.dedicated }} cluster from the CockroachDB {{ site.data.products.cloud }} Console. This page guides you through the process for an Ad ...
```
```suggestion
Now that [CockroachDB {{ page.page_version }}](https://www.cockroachlabs.com/docs/releases/ {{ page.page_version }}) is available, an [Org Administrator]({% link cockroachcloud/authorization.md %}#org-administrator) can upgrade your CockroachDB {{ site.data.products.dedicated }} cluster from the CockroachDB {{ site.data.products.cloud }} Console. This page shows how to upgrade a cluster in CockroachDB Cloud to {{ page.page_version }}.
```

mdlinville commented on src/current/cockroachcloud/upgrade-to-v24.1.md:
```diff
@@ -0,0 +1,174 @@
+Upgrading a CockroachDB {{ site.data.products.dedicated }} cluster to a new major version is opt-in. Before proceeding, review the CockroachDB {{ site.data.products.cloud }} [CockroachDB Cloud Upgrade Policy](https://cockroachlabs.com/docs/cockroachcloud/upgrade-policy).
+{{site.data.alerts.end}}
+
+This page shows how to upgrade a cluster in CockroachDB Cloud to {{ page.page_version }}.
```
```suggestion
```

rmloveland commented:
pre-review questions/comments: 

1. I guess `upgrade-to-v24.1.md` is mostly a copy of the previous version's file ?  I'd like to know before I bother reading this whole huge file 
2. the render at https://deploy-preview-18427--cockroachdb-docs.netlify.app/docs/releases/cloud.html#april-2-2024 is pretty busted, it's in like h2 font
3. the render at https://deploy-preview-18427--cockroachdb-docs.netlify.app/docs/cockroachcloud/upgrade-to-v23.2.html?filters=single-node#prepare-for-brief-unavailability is showing a double copy of the text you're editing on that page.  Just a rendering bug?

mdlinville commented:
> pre-review questions/comments:
> 
> 1. I guess `upgrade-to-v24.1.md` is mostly a copy of the previous version's file ?  I'd like to know before I bother reading this whole huge file

It is. I'll edit the commit history to show the state of the file after it was copied but before it was edited.

> 2. the render at https://deploy-preview-18427--cockroachdb-docs.netlify.app/docs/releases/cloud.html#april-2-2024 is pretty busted, it's in like h2 font
Thanks, I'll figure this out.

> 3. the render at https://deploy-preview-18427--cockroachdb-docs.netlify.app/docs/cockroachcloud/upgrade-to-v23.2.html?filters=single-node#prepare-for-brief-unavailability is showing a double copy of the text you're editing on that page.  Just a rendering bug?

Thanks, I'll figure this out too.

mdlinville commented:
> pre-review questions/comments:
> 
> 1. I guess `upgrade-to-v24.1.md` is mostly a copy of the previous version's file ?  I'd like to know before I bother reading this whole huge file

I've reorganized the commit history so you can see exactly what's different from the v23.2 file now. Hope it helps.

> 2. the render at https://deploy-preview-18427--cockroachdb-docs.netlify.app/docs/releases/cloud.html#april-2-2024 is pretty busted, it's in like h2 font

Fixed by removing some extra whitespace and a line that just had a `-` in it. I don't know why that caused this behavior.

> 3. the render at https://deploy-preview-18427--cockroachdb-docs.netlify.app/docs/cockroachcloud/upgrade-to-v23.2.html?filters=single-node#prepare-for-brief-unavailability is showing a double copy of the text you're editing on that page.  Just a rendering bug?

Nope, literally the same paragraph twice in a row. Fixed in both the v23.2 and v24.1 file, verified the bug was not in v23.1.

rmloveland commented on src/current/cockroachcloud/upgrade-to-v24.1.md:
```diff
@@ -0,0 +1,170 @@
+
+### Reset SQL statistics
+
+Before upgrading to CockroachDB {{ page.page_version }}, it is recommended to reset the cluster's SQL statistics. Otherwise, it may take longer for the upgrade to complete on a cluster with large statement or transaction statistics tables. This is due to the addition of a new column and a new index to these tables. To reset SQL statistics, issue the following SQL command:
```
'SQL statistics' could be a link to https://www.cockroachlabs.com/docs/stable/cost-based-optimizer#table-statistics

mdlinville replied:
```suggestion
Before upgrading to CockroachDB {{ page.page_version }}, it is recommended to reset the cluster's [SQL statistics]({% link {{ page.page_version }}/cost-based-optimizer.md %}#table-statistics). Otherwise, it may take longer for the upgrade to complete on a cluster with large statement or transaction statistics tables. This is due to the addition of a new column and a new index to these tables. To reset SQL statistics, issue the following SQL command:
```

rmloveland commented on src/current/cockroachcloud/upgrade-to-v24.1.md:
```diff
@@ -0,0 +1,170 @@
+Your single-node cluster will be unavailable for a few minutes while the node is stopped and restarted with CockroachDB {{ page.page_version }}.
+</section>
+
+After it is started, an upgrade cannot be cancelled. Instead, you can wait for the upgrade to finish, then [roll it back back](#roll-back-the-upgrade) for up to 72 hours, after which time it will be finalized and cannot be rolled back.
```
typo: "roll it back back"

mdlinville replied:
```suggestion
After it is started, an upgrade cannot be cancelled. Instead, you can wait for the upgrade to finish, then [roll it back](#roll-back-the-upgrade) for up to 72 hours, after which time it will be finalized and cannot be rolled back.
```

rmloveland commented on src/current/cockroachcloud/upgrade-to-v24.1.md:
```diff
@@ -0,0 +1,170 @@
+
+### Expect temporary limitations
+
+Most {{ page.page_version }} features can be used right away, but some will be enabled only after the upgrade has been finalized. Attempting to use these features before finalization will result in errors:
```
typo: this sentence ends in a colon but is not followed by a list of things.  Suggest ending with a period since not followed by list

mdlinville replied:
Yeah this is currently an empty list until we have the info for v24.1. 
```suggestion
Most {{ page.page_version }} features can be used right away, but some will be enabled only after the upgrade has been finalized. Attempting to use these features before finalization will result in errors. {% comment %}TODO: List of temporary limitations for GA{% endcomment %}
```

rmloveland reviewed (approved):
LGTM, had a couple small suggestions / typos
